nixos/modules/services/networking/consul.nix

```diff
@@ -13,7 +13,14 @@ let
     };
   } // cfg.extraConfig;
 
+  jsonFormat = pkgs.formats.json { };
+  enabledServices = attrValues (filterAttrs (_: svc: svc.enable) cfg.services);
+  servicesCfg = jsonFormat.generate "consul-services.json" {
+    services = map (filterAttrs (k: _: k != "enable")) enabledServices;
```

Comment:
I'll note that this can be a bit of a footgun. Consul will fail to start if there's a service config that doesn't match its expectations. There's maybe no way around this, but anybody coming later to add an option that consul doesn't like will need to exclude it here too.

Reply:
I'm not deeply attached to having the `enable` property if we're worried that sets a precedent for adding Nix-only options here, but I'm not sure how likely that is tbh.